lib/backend/k8s/k8s_test.go

```diff
@@ -1967,6 +1987,222 @@ var _ = testutils.E2eDatastoreDescribe("Test Watch support", testutils.Datastore
 		ctx = context.Background()
 	})
 
+	Describe("watching Profiles", func() {
+		createTestServiceAccount := func(name string) {
+			sa := k8sapi.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+				},
+			}
+			_, err := c.ClientSet.CoreV1().ServiceAccounts("default").Create(&sa)
+			Expect(err).NotTo(HaveOccurred())
+		}
+		deleteAllServiceAccounts := func() {
+			var zero int64
+			err := c.ClientSet.CoreV1().ServiceAccounts("default").DeleteCollection(&metav1.DeleteOptions{GracePeriodSeconds: &zero}, metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+		BeforeEach(func() {
+			createTestServiceAccount("test-sa-1")
+			createTestServiceAccount("test-sa-2")
+		})
+		AfterEach(func() {
+			deleteAllServiceAccounts()
+		})
+		It("supports watching a specific profile (from namespace)", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: "namespace-1", Kind: apiv3.KindProfile}, "")
+			Expect(err).NotTo(HaveOccurred())
+			event := ExpectAddedEvent(watch.ResultChan())
+			Expect(event.New.Key.String()).To(Equal("Profile(kns.namespace-1)"))
+		})
+		It("supports watching a specific profile (from serviceAccount)", func() {
+			watch, err := c.Watch(ctx, model.ResourceListOptions{Name: "test-sa-1", Kind: apiv3.KindProfile}, "")
```

Comment:
I think in this case, we're passing in the wrong thing for the name. Basically, the name we receive on the output event should match the name we pass in.

In this case, that means the watch needs to parse the name and decide which of the two objects it actually is meant to watch (or both, if no name is present).

I think the tests need to be updated to pass in `kns.namespace-1` and `ksa.default.test-sa-1`, and then it should be more apparent what needs to be done here.

Sorry I didn't spot this earlier, but the names we pass in on this API are the _calico_ resource names, not the names of the backing Kubernetes objects.